msteams-platform/sbs-botbuilder-conversation-AI.yml

```diff
@@ -0,0 +1,709 @@
+            import path from "path";
+            ```
+
+            You need to import the required classes to use the functions and variables available in the class to build your app. You can leverage pre-built templates, and simply add your business logic to this scaffold.
+
+        1. Add the following code to create planner, moderator, and promptManager:
+
+            ```typescript
+            // Create AI components
+            const planner = new OpenAIPlanner({
+            apiKey: config.openAIKey,
+            defaultModel: 'text-davinci-003',
+            logRequests: true
+            });
+            const moderator = new OpenAIModerator({
+            apiKey: config.openAIKey,
+            moderate: 'both'
+            });
+            const promptManager = new DefaultPromptManager(path.join(__dirname, './prompts' ));
+            ````
+            * Planner: The `OpenAIPlanner` is a class that implements an interface called `planner`. The planner calls the LLM model 'text-davinci-003' for text completion. You can use the `OpenAIPlanner` or Azure OpenAI planner. If your using the Azure OpenAI planner, ensure that you define the endpoint property.
+
+            * PromptManager: `promptManager` Allows you to register your prompts, load a prompt template into the AI component. All the functions related to the prompt manager are available in the `DefaultPromptManager`.  The prompt manager parses the prompts defined in the `prompt.txt` folder. You can replace or override your own functionality in the prompt manager. 
+
+            * Moderator: The `OpenAIModerator` reviews both user's input and the model output to detect harmful speech, such as hate speech, self-harm, and violence.
+
+
+        1. Add the following code to define storage and application:
+
+            ```typescript
+            // Define storage and application
+            const storage = new MemoryStorage();
+            const app = new Application<ApplicationTurnState>({
+                storage,
+                ai: {
+                    planner,
+                    moderator,
+                    promptManager,
+                    prompt: 'chat',
+                    history: {
+                        assistantHistoryType: 'text'
+                    }
+                }
+            });
+            ```
+
+            * The `Application` class replaces the Teams Activity Handler class from Bot Framework SDK. The application class has all the information and bot logic required for an app. You can register actions or activity handlers for the app in this class.
+            * The `MemoryStorage` class is imported from the Bot Builder SDK and has a predefined logic to handle any issues when two or more events exist at the same time so that you don't want to worry about handling storage issues and focus on building your application.
```

Comment:
We want to keep it simple and give a simple definition of what the memory storage does. 

Replace with:

The `MemoryStorage` class is a memory based storage provider that stores and retrieves plan old JSON objects. It is imported from the Bot Builder SDK.

Reply:
Updated.